src/inc/msquic.h

```diff
@@ -517,6 +517,38 @@ typedef enum QUIC_PERFORMANCE_COUNTERS {
     QUIC_PERF_COUNTER_MAX,
 } QUIC_PERFORMANCE_COUNTERS;
 
+typedef struct QUIC_VERSION_SETTINGS {
+    union {
+        uint64_t IsSetFlags;
+        struct {
+            uint64_t DesiredVersionsList                    : 1;
+            uint64_t VersionNegotiationExtEnabled           : 1;
+            uint64_t RESERVED                               : 62;
+        } IsSet;
+    };
+
+    //
+    // Flags first so we can use offset to get versioned length of struct.
+    //
+    uint8_t VersionNegotiationExtEnabled    : 1;
+    uint8_t RESERVED                        : 7;
+    const uint32_t* DesiredVersionsList;
+    uint32_t DesiredVersionsListLength;
```

Comment:
I guess we could also do `const uint32_t* Versions` but I do like the idea of a contiguous buffer to make "get" simple.

Reply:
That makes it so this struct can never be extended. Which if we move the VN extension flag back into settings, I'd probably be ok with.

Reply:
Yeah, I'm not necessarily opposed to making this struct set in stone, so long as we can get it right... One option is we wrap it in some kind of `QUIC_PREVIEW_FEATURES` define to allow us to modify it in the future just in case the version negotiation draft changes before it goes RFC. Thoughts?